packages/haul-cli/src/commands/init.ts

```diff
@@ -1,5 +1,327 @@
+
+  await delay(1000);
+
+  let project = fs
+    .readFileSync(path.join(xcodeProject, 'project.pbxproj'))
+    .toString();
+
+  const haulSignature = 'added by Haul';
+
+  /* Make sure we check both project iOS and iOS-TV, that's the magic behind "2" constant */
+
+  const PROJECTS_COUNT = 2;
+
+  const countOccurrences = (search: string) => {
+    return project.split(search).length - 1;
+  };
+
+  // Are we already integrated?
+  if (countOccurrences(haulSignature) === PROJECTS_COUNT) {
+    progress.info('Haul is already part of your build scripts');
+    return;
+  }
+
+  const originalTask = 'shellScript = "export NODE_BINARY=node';
+
+  if (countOccurrences(originalTask) !== PROJECTS_COUNT) {
+    progress.warn(
+      `Couldn't edit Xcode project. Haven't recognized 'Bundle React Native code and images' build phase.`
+    );
+    return;
+  }
+
+  const haulTask = `shellScript = "# ${haulSignature}\\nexport CLI_PATH=node_modules/@haul-bundler/cli/bin/haul.js\\nexport NODE_BINARY=node`;
+
+  project = project.replace(new RegExp(originalTask, 'g'), haulTask);
+
+  fs.writeFileSync(path.join(xcodeProject, 'project.pbxproj'), project);
+  progress.succeed('Added haul to your Xcode build scripts');
+}
+
+async function modifyGradleBuild(cwd: string) {
+  let gradleBuildFile;
+  // Does `android/app/build.gradle` exist?
+  const androidPath = path.join(cwd, 'android/app');
+  if (fs.existsSync(androidPath)) {
+    const list = fs
+      .readdirSync(androidPath)
+      .filter(file => file.includes('build.gradle'));
+    // Do nothing if multiple projects were found
+    if (list.length === 1) {
+      gradleBuildFile = path.join(androidPath, list[0]);
+    }
+  }
+  // Otherwise, ask for path to a file
+  if (!gradleBuildFile) {
+    const result = (await inquirer.prompt([
+      {
+        type: 'input',
+        name: 'entry',
+        message: 'Enter path to the android/app/build.gradle file',
+        validate: (pathToFile: string) =>
+          fs.existsSync(pathToFile) && pathToFile.includes('build.gradle')
+            ? true
+            : `${pathToFile} is not a valid build.gradle`,
+      },
+    ])) as { entry: string };
+    gradleBuildFile = path.resolve(result.entry);
+  }
+  const progress = ora('Adding haul to your build.gradle');
+  await delay(1000);
+  let project = fs.readFileSync(gradleBuildFile).toString();
+  const cliString = '"node_modules/@haul-bundler/cli/bin/haul.js"';
+
+  // Are we already integrated?
+  if (project.includes(cliString)) {
+    progress.info('Haul is already part of your build.gradle');
+    return;
+  }
+  project = project.replace(
+    /project\.ext\.react = \[\n(.+)\n\]/,
+    dedent`
+    project.ext.react = [
+    $1
+        cliPath: ${cliString}
+    ]
+    `
+  );
+  fs.writeFileSync(gradleBuildFile, project);
+  progress.succeed('Added haul to your build.gradle');
+}
+
+function getRunScript(scriptName: string) {
+  const runCommand = scriptName === 'start' ? 'yarn' : 'yarn run';
```

Comment:
`yarn` and `yarn run` are equivalent in terms of running binaries/commands

Reply:
There is but not in this case